subprojects/persistent-cache/src/main/java/org/gradle/cache/internal/DefaultFileLockManager.java

```diff
@@ -146,6 +145,9 @@ public DefaultFileLock(File target, LockOptions options, String displayName, Str
             LockStateSerializer stateProtocol = options.isUseCrossVersionImplementation() ? new Version1LockStateSerializer() : new DefaultLockStateSerializer();
             lockFileAccess = new LockFileAccess(lockFile, new LockStateAccess(stateProtocol));
             try {
+                if (whenContended != null) {
```

Comment:
Yes the tests are all added in a different commit. Without this change, the _whenContended_ action is only registered after the lock is acquired. And this means that the lock does not know that it is _allowed to be contended_ and thus will not listen to confirmation messages. 

This test fails without this:
https://github.com/gradle/gradle/blob/0913232e8536330b6087ba0f6f90f5edc6ebc600/subprojects/persistent-cache/src/integTest/groovy/org/gradle/cache/internal/DefaultFileLockManagerContentionIntegrationTest.groovy#L98-L117